File: aoj/volume100/java/A10033.java
import java.util.ArrayList;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.List;
import java.util.Scanner;

class A10033 {
    public static void main(String[] args) {
        List<Deque<String>> stacks = new ArrayList<Deque<String>>();

        Scanner scan = new Scanner(System.in);

        int nStacks = scan.nextInt();
        for (int i = 0; i < nStacks; i++) {
            stacks.add(new ArrayDeque<String>());
        }

        while (true) {
            String instruction = scan.next();
            String block;
            int src, dst;

            switch (instruction) {
            case "quit":
                return;

            case "push":
                dst   = scan.nextInt() - 1;
                block = scan.next();
                stacks.get(dst).push(block);
                break;

            case "pop":
                src   = scan.nextInt() - 1;
                block = stacks.get(src).pop();
                System.out.println(block);
                break;

            case "move":
                src   = scan.nextInt() - 1;
                dst   = scan.nextInt() - 1;
                block = stacks.get(src).pop();
                stacks.get(dst).push(block);
                break;
            }
        }
    }
}

File: aoj/volume100/ruby/A10033.rb
stacks  = []
n_stack = gets.to_i
n_stack.times do |i|
  stacks[i] = []
end

while line = gets do
  operation, *args = line.chomp.split
  case operation
  when "push"
    dst, block = args[0].to_i-1, args[1]
    stacks[dst].push(block)
  when "pop"
    src = args[0].to_i-1
    puts stacks[src].pop()
  when "move"
    src, dst = args[0].to_i-1, args[1].to_i-1
    block = stacks[src].pop()
    stacks[dst].push(block)
  when "quit"
    break
  end
end
